server/block/explosion.go

```diff
@@ -0,0 +1,164 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/cube/trace"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+	"math/rand"
+	"time"
+)
+
+var rays = make([]mgl64.Vec3, 0, 1352)
+
+func init() {
+	for x := 0.0; x < 16; x++ {
+		for y := 0.0; y < 16; y++ {
+			for z := 0.0; z < 16; z++ {
+				if x != 0 && x != 15 && y != 0 && y != 15 && z != 0 && z != 15 {
+					continue
+				}
+				rays = append(rays, mgl64.Vec3{x/15*2 - 1, y/15*2 - 1, z/15*2 - 1}.Normalize().Mul(0.3))
+			}
+		}
+	}
+}
+
+type ExplodableEntity interface {
+	Explode(c ExplosionConfig, impact float64)
+}
+
+type Explodable interface {
+	Explode(pos cube.Pos, c ExplosionConfig)
+}
+
+type ExplosionConfig struct {
+	World *world.World
+	Pos   mgl64.Vec3
+	Size  float64
+	Rand  rand.Source
+	Fire  bool
```

Comment:
Would maybe make this a little less short, maybe something like `CreateFire` or `SpawnFire`.